pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -597,6 +599,9 @@ private boolean processStreamEvents(MessageBatch messagesAndOffsets, long idlePi
           realtimeRowsDroppedMeter =
               _serverMetrics.addMeteredTableValue(_clientId, ServerMeter.REALTIME_ROWS_FILTERED,
                   reusedResult.getSkippedRowCount(), realtimeRowsDroppedMeter);
+          if (_trackFilteredMessageOffsets) {
+            _filteredMessageOffsets.add(messageOffset.toString());
```

Comment:
Why are we tracking individual offsets? Wouldn't tracking the number of messages filtered be enough?